### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by meteorlxy** [vuepress/core#68] [General]
## IMPORTANT

[Vue team has decided to focus on VitePress as the main recommended SSG in the long run](https://vitepress.dev/guide/what-is-vitepress#what-about-vuepress). So the roadmap has been delayed a lot.

Future of this project:
- VuePress will transform from a @vuejs project to a community project, and will continue to be maintained by current team.
- VitePress will keep its current name and would NOT become VuePress v3.

---

### Alpha Milestone

- [x] vite support (2021-03)

---

### Features

- [x] plugin-prismjs (2021-03)
- [x] plugin-shiki (2021-03)
- [x] Markdown code snippets #15 (2021-04)
- [x] Default theme dark mode (including migration of css vars) #29 (2021-05)
- [x] plugin-search (built-in search of v1) #35 (2021-04)
- [x] theme template support #387 (2022-01)
- [x] add devtools support and remove plugin-debug (2022-01)
- [x] allow html title customization #338
- [ ] Client config improvement
- [ ] SSR normalization

### Documentation

- [x] Migration guide (2021-03)
- [x] Node API (2021-08)
- [x] Cookbook (continually updated, accepting new recipes)
- [x] V2 Awesome VuePress #735 (@Mister-Hope)
---

### Core features built by Community

- [x] [blogging](http://plugin-blog2.vuejs.press/) (@Mister-Hope )
- [x] [feed](https://plugin-feed2.vuejs.press/) (@Mister-Hope)
- [x] [extended markdown features](https://plugin-md-enhance.vuejs.press/) (@Mister-Hope)
- [x] [seo](https://plugin-seo2.vuejs.press/) (@Mister-Hope )
- [x] [sitemap](https://plugin-sitemap2.vuejs.press/) (@Mister-Hope )
- [x] [Redirects](https://plugin-redirect.vuejs.press/) (@Mister-Hope )
- [x] [Enhanced PWA](https://plugin-pwa2.vuejs.press/)  (@Mister-Hope)
- [x] [Copy button for code blocks](https://plugin-copy-code2.vuejs.press/) (@Mister-Hope)
- [x] [Copyright info](https://plugin-copyright2.vuejs.press/) (@Mister-Hope)
- [x] [Image viewer](https://plugin-photo-swipe.vuejs.press/)  (@Mister-Hope)

### Community Themes

- [x] theme supporting all basic features: vuepress-theme-hope(@Mister-Hope )

Please leave a comment if you want to contribute to ensure that you have no conflicts with other developers.

**Comment by selected-pixel-jameson** [comment]
What's the target for a stable release? There doesn't seem to be much left outside of the default dark theme in the roadmap features?

**Reply by meteorlxy** [reply]
A stable release means that v2 API is solid.

Up to now, v2 API is almost stable. So the remaining thing is to optimize and add some reasonable new features.

Hope we can release VuePress v2 before or at end of 2021 Q2, after which vue3 will become the default version of `vue` package.

**Reply by blackfyre** [reply]
Eagerly waiting for the new version 😁

**Reply by Benbinbin** [reply]
wait for the awesome project to be stable

**Reply by kid1412621** [reply]
1y later😂

**Reply by kaikaibenkai** [reply]
2y later... What happened to v2?

**Reply by fdevans** [reply]
Also interested in a timeline for a stable release.  Completely understand the challenges of building in a fast moving technology environment, but it would be good to have something that feels more official to use than the Beta releases.

**Reply by meteorlxy** [reply]
@kaikaibenkai - See the "IMPORTANT" section at the top. I thought people may read from the first line 🤦‍♂️ 

@fdevans - Waiting for vite v5 and then we may enter rc stage.

**Reply by kaikaibenkai** [reply]
Since VitePress won't evolve into VuePress3, v2 is a hero for Vuepress😢

**Comment by loctran016** [comment]
It will be nice if vuepress can **create pages from data** (you can find a similar function at [11ty](https://www.11ty.dev/docs/pages-from-data/))
For example, I have a ssg.json:

```json
{
"vuepress": {
"Title": "vuepress",
"Description": "A vue-based SSG"
},
"gatsby": {
"Title": "gatsby",
"Description": "A react-based SSG"
}
```
Will create 2 files gatsby.md and vuepress.md with following content
```md
---
title: vuepress
description: A vue-based SSG
---
```
```md
---
title: gatsby
description: A react-based SSG
---
```

**Reply by meteorlxy** [reply]
It will be nice if users can always read the docs 😢 

https://vuepress2.netlify.app/advanced/cookbook/adding-extra-pages.html

**Reply by loctran016** [reply]
Really sorry to you

**Comment by Mister-Hope** [comment]
Just be curious, why use `--c-` prefix in `@vuepress/theme-default`, what does `c` mean for short?

I think changing to `--vuepress-` or just `--vp-` will be better. Most of the plugins will choose to follow the variable which the default theme uses. So it would be nice to have a short and clear prefix

**Reply by meteorlxy** [reply]
`--c-` for `--color-`

**Reply by Mister-Hope** [reply]
Got it.

**Comment by dumptyd** [comment]
I was trying this out in a new project and ran into this error (https://github.com/csstree/csstree/issues/162) because of the usage of csso plugin for postcss here: 

https://github.com/vuepress/vuepress-next/blob/main/packages/@vuepress/bundler-webpack/src/config/handleModuleStyles.ts#L89

Would it be possible (or more specifically a priority) to fix this on Vuepress' side by using another CSS optimizer or making the plugin opt-in? I know I can tap into the webpack config and remove that plugin but I think that solution is too intrusive.

**Comment by KnorpelSenf** [comment]
The roadmap looks pretty complete to me. What is there left to do before it will leave the beta?

**Comment by straumat** [comment]
If you are impatient as I am and want to use it as soon as possible, I used the 2 beta for my open source project and it works great: [https://trading-bot.cassandre.tech/](https://trading-bot.cassandre.tech/)

**Comment by bronzelegs** [comment]
I'd like to help

pull doc tree from repository
oauth / membership
redis datastore
commenting (disqus intergration)
social functions (like, private message, block)
shopping cart interface

this will require other vues, maybe plugins, and a datastore interface (node)
I'd like to use redis or a triple store maybe for all the data

theres little doc on how to do these things, how can I help?

tdavis0525#gmail.com I dont know what account github has

t want to use it as a component of a larger project as the display interface for the content, then bunde it all with docker and add control interfaces

**Comment by zgayjjf** [comment]
Is there any plans to upgrade [vuepress-community](https://github.com/vuepress/vuepress-community/issues/101)

**Reply by Mister-Hope** [reply]
We may have plans to do that, after we split vuepress/vuepress-next to vuepress/core and vuepress/ecosystem

Meteorlxy is a bit busy now. And I am at school

**Reply by zgayjjf** [reply]
I can help with that if you can guide me on where to start.

**Reply by Mister-Hope** [reply]
You can fork that repo and try making plugins compatable in your forked repo. After we split the repo, you can make pr to add each plugins in to the new repo.

**Comment by nagi1** [comment]
Hey, What is messing to release first official version?

I want to contribute 🚀

**Reply by qqlcx5** [reply]
I developed a theme based on the default version, with categories, timelines, and tabs

url: https://feyoudao.cn/vuepress-next/

Hope to have enthusiast can offer a point of opinion

**Comment by bravecorvus** [comment]
Hey is there a way to do "logo" in config.js like in v1?

**Comment by jortfal** [comment]
Hi @meteorlxy, 

I am planning to create a documentation site with VuePress v2.0.0.

But in March last year I commented that I intended to transfer the project to the community because the Vue.js team is considering replacing VuePress with VitePress.

What is the future of VuePress?
Will it be replaced by Vite?

It is important to know these answers in order to bet on one project or the other. 

Thank you.

Best regards.

**Reply by jortfal** [reply]
Hi @yyx990803,

Do you have any information regarding my questions above?

Thank you.

Best regards.

**Reply by meteorlxy** [reply]
VuePress v2 will be continuously maintained by the current team, but might not be a project of @vuejs organization. (and might be renamed)

For now you can choose to use either project as you like:
- VuePress v2 - a community project
- VitePress - a @vuejs project

**Reply by jortfal** [reply]
Hi @meteorlxy,

thank you very much for your reply. 

I have tried VitePress and I loved it. I am creating a private documentation website with this tool.

Regards.

**Comment by CanadianBeaver** [comment]
Can somebody tell the approximate date when VuePress v2 or VitePress v1 will be ready for stable release?

**Comment by myServiceAccount** [comment]
👍✅<a href="#"><img src="https://camo.githubusercontent.com/546b4aa458ac9eeeaa55acb791a8d776692b5840ff49fe112e95900bee4afe39/68747470733a2f2f6b6f6d617265762e636f6d2f67687076632f3f757365726e616d653d53616d69725061756c31"  width="1" height="1"></a>

**Comment by myServiceAccount** [comment]
<a href="#"><img src="https://camo.githubusercontent.com/546b4aa458ac9eeeaa55acb791a8d776692b5840ff49fe112e95900bee4afe39/68747470733a2f2f6b6f6d617265762e636f6d2f67687076632f3f757365726e616d653d53616d69725061756c31"  width="1" height="1"></a>

**Comment by liudonghua123** [comment]
So is it recommend to use https://github.com/vuejs/vitepress instead of vuepress in 2023?

**Reply by Mister-Hope** [reply]
See:

- https://vitepress.dev/guide/what-is-vitepress#what-about-vuepress
- https://github.com/vuepress/vuepress-next/discussions/1281
- https://github.com/vuepress/vuepress-next/discussions/1281#discussioncomment-5388699

**Comment by CanadianBeaver** [comment]
What about the release date of someone of vite/vue(press) without comparison?
